src/Farmer/Builders/Builders.WebApp.fs

```diff
@@ -736,6 +736,13 @@ type WebAppBuilder() =
     member this.AddCustomDomains(state, customDomains:string list) = customDomains |> List.fold (fun state domain -> this.AddCustomDomain(state, domain)) state
     member this.AddCustomDomains(state, domainConfigs:DomainConfig list) = domainConfigs |> List.fold (fun state domain -> this.AddCustomDomain(state, domain)) state
     member this.AddCustomDomains(state, customDomainsWithThumprint:(string * ArmExpression) list) = customDomainsWithThumprint |> List.fold (fun state domain -> this.AddCustomDomain(state, domain)) state
+    
+    /// Map specified port traffic from your docker container to port 80 for App Service
+    [<CustomOperation "docker_port">]
+    member _.DockerPort(state: WebAppConfig, dockerPort:int) = { state with DockerPort = Some dockerPort }
+    /// Enables the zone redundancy in service plan
+    [<CustomOperation "zone_redundant">]
+    member this.ZoneRedundant(state:WebAppConfig, flag:FeatureFlag) = {state with ZoneRedundant = Some flag}
```

Comment:
Does this only work for webapps? What about functions?

Reply:
ZoneRedundancy is an app service plan setting. When farmer creates a function if asp is not specified it creates the one with Dynamic SKU (for some reason hardcoded) which doesn't support zone redundancy. I have noticed that in web apps we can specify SKU, so added also ZoneRedundancy support. But we shouldn't be worried about it as we never create asp as part of the web app or function creation. In our circumstances, we will specify ZoneRedundant when we create asp. Then link web app/functions to use it. That part was already merged.